### PR TITLE
[SON-202] Implement cli_wallet commands for maintenance mode

### PR DESCRIPTION
**Pull request by gladcow**
Also fixes a bug with going SON to active mode. Adds `start_son_maintenance` and `stop_son_maintenance` calls to CLI wallet and tests for this functions.

### CODE REVIEW

**Comment by bobinson** [comment]
@satyakoneru  - needs your review on this.

File: libraries/chain/db_maint.cpp [hunk 1]
```diff
@@ -471,6 +471,18 @@ void database::update_active_sons()
    } else {
       ilog( "Active SONs set CHANGED" );
       // Store new SON info, initiate wallet recreation and transfer of funds
+      for( const son_object& son : cur_active_sons )
```
Comment (satyakoneru) [inline libraries/chain/db_maint.cpp]: There is a requirement in FR document wherein if a SON is in maintenance mode for more than 12 hours then it should be deregistered by witnesses. If we move all active SONs to inactive we might lose track of the maintenance mode.

We might also have to change the heartbeat evaluator code to check for the state,
i.e. if a SON is in maintenance mode and it sends heartbeat we shouldn't move it straight back to active. We should check if that SON is on the list of active sons in GPO. (gpo.active_sons) after chain maintenance.

Comment (gladcow) [reply libraries/chain/db_maint.cpp]: Good catch.) I'll add the separate commit to fix it.

**Comment by gladcow** [comment]
@satyakoneru ,  I've fixed an issue with lost maintenance state, take a look one more time,  please.

**Comment by gladcow** [comment]
Contains compilation error,  don't merge it now, please.

**Comment by gladcow** [comment]
It looks like smth wrong with CI,  it uses incorrect `fc` commit, local build has not errors. In any case it is better to merge this PR after the https://github.com/peerplays-network/peerplays/pull/264 merging, I'll rebase this PR on it to adapt test changes.

**Comment by bobinson** [comment]
@RoshanSyed  : please check the CI errors.

**Comment by gladcow** [comment]
@bobinson ,  re-review, please, I've rebased the PR and now CI is green.

**Pull request merged**